core/src/main/java/io/questdb/griffin/engine/ops/OperationDispatcher.java

```diff
@@ -77,6 +77,9 @@ public OperationFuture execute(
                     closeOnDone
             );
             return future;
+        } catch (Throwable th) {
+            operation.close();
```

Comment:
there are two `close()` calls in `catch` and `finally` with nothing seemingly guarding them

Reply:
What kind of guard do we need?